### ISSUE TITLE
Converting file to csv stream is adding new line

### ISSUE THREAD

**Issue by raghu64** [SheetJS/sheetjs#1553]
I am trying to convert the xlsx file to csv stream and when i listen on `data` and print the chuck i'm seeing a new empty line which is not present in the file

```js
var xlsx = require('xlsx')
const wb = xlsx.readFile('testData.xlsx')
const ws = wb.Sheets[wb.SheetNames[0]];
const stream = xlsx.stream.to_csv(ws)
stream.on('data', chunk => {
    console.log(chunk)
    console.log(chunk.toString())
    console.log(chunk.toString().length)
})
```

the `testData.xlsx` file contains the following data
```
a,b,c
1,2,3
```
When i run the above mentioned code i'm getting the following output
```
<Buffer ef bb bf>

1
<Buffer 61 2c 62 2c 63 0a>
a,b,c

6
<Buffer 31 2c 32 2c 33 0a>
1,2,3

6
```
in which we can see a new line.

Can someone help me understand why xlsx is adding that new line and how to read the correct data from the file?

**Comment by SheetJSDev** [comment]
The newline characters are written when the individual rows are written.  As a result, it currently writes a newline character at the end.

If you would like to change that, https://github.com/SheetJS/sheetjs/blob/master/bits/97_node.js#L27 should keep track of whether the first line was written:  It would look like:

```diff
		var BOM = false;
+		var wrote_first_line = false;
// ...
-					stream.push(row + RS);
+					stream.push(wrote_first_line ? RS + row : row);
+					wrote_first_line = true;
					break;
```

**Comment by Rnamita** [comment]
I would like to work on this. Apart from above mentioned code changes, there has to be additional code to remove the newlines as well I believe? Guidance in this regard would be helpful as this is my first open source contribution

**Comment by SheetJSDev** [comment]
Fixed in a32b30414b3a1da61ba6d6b847a473f54481815a . CSV export (both stream and normal) omits trailing record separator
